### PR TITLE
Feat: Discord Private Notifications

### PR DESCRIPTION
**Pull request by JayeshVP24**
## Screenshots (if appropriate)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,9 +92,10 @@
     "@types/jsonwebtoken": "^9.0.6",
     "@types/lint-staged": "~13.3.0",
     "@types/lodash.debounce": "~4.0.9",
-    "@types/node": "20.14.2",
-    "@types/react": "18.3.3",
-    "@types/react-dom": "18.3.0",
+    "@types/node": "20.11.0",
```
[the remark this answers is not in the document]
Comment (JayeshVP24) [reply package.json:95]: dont know what made it, but resetting

File: public/clean-skills.csv [hunk 2]
```diff
@@ -0,0 +1,414 @@
+id,editedSkills
```
[the remark this answers is not in the document]
Comment (JayeshVP24) [reply public/clean-skills.csv:1]: that was a one time task, deleting the file

**Comment by JayeshVP24** [comment]
done, @a20hek 
moved types/papaparse to devDeps instead of deps
i used papaparse in a cleanup task

File: src/pages/api/listings/announce/[id].ts [hunk 3]
```diff
@@ -87,8 +88,15 @@ async function announce(req: NextApiRequestWithUser, res: NextApiResponse) {
...
       },
+      include: {
+        sponsor: true,
+      },
     });
-
+    try {
+      await discordWinnersAnnouncement(result);
+    } catch (err) {
+      console.log('Discord Listing Update Message Error', err);
```
Comment (a20hek) [inline src/pages/api/listings/announce/[id].ts:98]: use logger.error (news: we have a new logging system now)

File: src/pages/api/listings/create.ts [hunk 4]
```diff
@@ -70,6 +71,14 @@ async function handler(req: NextApiRequestWithUser, res: NextApiResponse) {
       include: { sponsor: true },
     });
 
+    try {
+      await discordListingUpdate(
+        result,
+        result.isPublished ? 'Published' : 'Draft Added',
+      );
+    } catch (err) {
+      console.log('Discord Listing Update Message Error', err);
```
Comment (a20hek) [inline src/pages/api/listings/create.ts:80]: logger

File: src/pages/api/listings/update/[id].ts [hunk 5]
```diff
@@ -121,8 +122,27 @@ async function bounty(req: NextApiRequestWithUser, res: NextApiResponse) {
...
+        await discordListingUpdate(result, 'Unpublished');
+      }
+      if (
+        listingBeforeUpdate.isPublished === false &&
+        result.isPublished === true
+      ) {
+        await discordListingUpdate(result, 'Published');
+      }
+    } catch (err) {
+      console.log('Discord Listing Update Message Error', err);
```
Comment (a20hek) [inline src/pages/api/listings/update/[id].ts:142]: logger

File: src/utils/scripts.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
+import * as fs from 'fs';
```
Comment (a20hek) [inline src/utils/scripts.ts:1]: this file has been committed accidentally too right?

**Review by a20hek — CHANGES_REQUESTED** [review]

File: src/utils/scripts.ts (hunk 6 above)
Comment (JayeshVP24) [reply src/utils/scripts.ts:1]: yes

**Review by a20hek — APPROVED** [review]

**Pull request merged**